js/src/components/google-combo-account-card/connect-mc/connect-mc.js

```diff
@@ -0,0 +1,105 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { useEffect, useState } from '@wordpress/element';
+import classNames from 'classnames';
+
+/**
+ * Internal dependencies
+ */
+import useGoogleMCAccount from '.~/hooks/useGoogleMCAccount';
+import ConnectAccountCard from '../connect-account-card';
+import ConnectMCBody from './connect-mc-body';
+import ConnectMCFooter from './connect-mc-footer';
+import SpinnerCard from '.~/components/spinner-card';
+import AccountConnectionStatus from '.~/components/google-mc-account-card/connect-mc/account-connection-status';
+import useConnectMCAccount from '.~/hooks/useConnectMCAccount';
+import { hasAccountConnectionIssue } from '.~/components/google-mc-account-card/connect-mc/utils';
+
+/**
+ * Clicking on the "Switch account" button to select a different Google Merchant Center account to connect.
+ *
+ * @event gla_mc_account_switch_account_button_click
+ * @property {string} context (`switch-url`|`reclaim-url`) - indicate the button is clicked from which step.
+ */
+
+/**
+ * ConnectMC component.
+ *
+ * This component renders Merchant Center connection section.
+ * It is using createMCAccount and resultCreateMCAccount from the parent component.
+ *
+ * @param {Object}   props
+ * @param {Function} props.createMCAccount Callback function for creating a new Merchant Center account.
+ * @param {Object}   props.resultCreateMCAccount The result of the create account request.
+ */
+const ConnectMC = ( { createMCAccount, resultCreateMCAccount } ) => {
+	const {
+		googleMCAccount,
+		hasFinishedResolution,
+		isReady: isGoogleMCReady,
+	} = useGoogleMCAccount();
+	const [ accountID, setAccountID ] = useState();
+	const [ handleConnectMC, resultConnectMC ] =
+		useConnectMCAccount( accountID );
+
+	useEffect( () => {
+		if ( isGoogleMCReady ) {
+			setAccountID( googleMCAccount.id );
+		}
+	}, [ googleMCAccount, isGoogleMCReady ] );
```

Comment:
Not sure I understand why we need to save the MC account separately as component state here rather than using `googleMcaccount.id` directly?

Reply:
@joemcgill We need to save the value set from the select in case the user needs to connect to a different acccount.